features/promotion/managing_coupons/being_unable_to_generate_too_many_coupons_with_prefix_and_suffix.feature

```diff
@@ -0,0 +1,45 @@
+@managing_promotion_coupons
+Feature: Being unable to generate too many coupons with prefix and suffix
+    In order to avoid generating duplicated coupons for my promotions
+    As an Administrator
+    I want to be prevented from generating too many coupons with the same prefix or suffix
+
```

Comment:
When I first read this scenario, I didn't understand why it's not possible to generate 2 more coupons if 8 are generated already.

What do you think about adding the following section before `Background`?

```gherkin
Rules:
    - Generated coupon code uses 16 characters (numbers from 0 to 9, letters from A to F)
    - To prevent guessing coupons codes, it is not allowed to generate more than 50% of possible coupons
```